python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -763,10 +765,19 @@ def _init(self, *args, **kwds):
         else:
             ## FIRST INITALIZE A RAW OBJECT INSTANCE CORRESPONDING TO 'pluginclass'
             ## Object was not passed by construction so need to construct new object for internal use
-            instance = pluginclass()
+            # Case 1 j = Job(myExistingJob)            # We want to perform a deepcopy
+            if len(args) == 1 and isinstance(args[0], pluginclass):
+                instance = deepcopy(stripProxy(args[0]))
+            # Case 2 file_ = LocalFile('myFile.txt')   # We need to pass the (stripped) arguments to the constructor only if the 
+            elif len(args) < len(getargspec(pluginclass.__init__)[0]):
```

Comment:
You're probably correct about letting any error be propagated but I'm wanting to protect against older broken classes for now and just warn the user which is less strict but relatively safe.